main.nf

```diff
@@ -417,3 +427,80 @@ if (params.step =~ 6){
     """
   }
 }
+
+if (params.step =~ 7){
+  if (params.rbindir) ch_rbin_dir = Channel.fromPath("${params.rbindir}")
+
+  process step_7_samplefile_gen {
+    label 'new_steps'
+    //echo true
+    publishDir "results/", mode: params.mode
+
+    input:
+    path rbin_dir from ch_rbin_dir
+
+    output:
+    path "samplefile.txt" into ch_samplefile
+
+    script:
+    """
+    cnest_dev.py step7 \
+      --samplefile samplefile.txt \
+      --rbindir $rbin_dir
+    """
+  }
+}
+
+if (params.step =~ 8){
+  if (params.rbindir) ch_rbin_dir = Channel.fromPath("${params.rbindir}")
+  if (params.samplefile) ch_samplefile = Channel.fromPath("${params.samplefile}")
+
+  process get_chr {
+    publishDir "results/${params.project}", mode: params.mode
+    label 'new_steps'
+
+    input:
+    path index from ch_index_tab
+
+    output:
+    path "chr_list.txt" into ch_chr_lits_file
+
+    script:
+    """
+    cut -f1 $index | uniq > chr_list.txt
+    """
+  }
+
+  ch_chr_lits_file
+    .map { it.text.split('\n').collect { it.toInteger() } }
```

Comment:
This implies that chromosome are only represented with a number, is that always the case here?

Reply:
Yes for the tool always accepts chr as numbers.